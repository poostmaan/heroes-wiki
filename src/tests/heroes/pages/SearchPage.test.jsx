import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter, useNavigate } from "react-router-dom"
import { SearchPage } from "../../../heroes"

const mockedUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUseNavigate
}))

describe('<Search Page /> testing', () => {
  test('should render correctly', () => {
    const {container} = render(
        <MemoryRouter>
            <SearchPage />
        </MemoryRouter>
    );

    // const component = screen.debug();

    expect( container ).toMatchSnapshot();

  })

  test('should render Batman info and save in an input the QueryParameters', () => {
    render(
        <MemoryRouter initialEntries={['search?q=Batman']}>
            <SearchPage />
        </MemoryRouter>
    );

    // const component = screen.debug(); 
    // screen.debug() 
    const input = screen.getByRole('textbox') 
    expect( input.value ).toBe('Batman')

    const img = screen.getByRole('img') 
    expect( img.src ).toContain('/assets/heroes/dc-batman.jpg')

    const alertNoFound = screen.getByLabelText('alert-nofound')
    expect( alertNoFound.getAttribute('style') ).toBe('display: none;')

  })

  test('should show a error whether a hero isn`t found', () => {
    render(
        <MemoryRouter initialEntries={['search?q=Batman123']}>
            <SearchPage />
        </MemoryRouter>
    );

    // screen.debug();

    const alertNoFound = screen.getByLabelText('alert-nofound')
    expect( alertNoFound.getAttribute('style') ).toBe(null)

  })

  test('should execute navigate in the new page', () => {
    render(
      <MemoryRouter >
          <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByRole('textbox') 
    const form = screen.getByRole('form');

    fireEvent.change( input , {target: {value: 'Superman'}});
    fireEvent.submit( form );
    // expect( input.value ).toBe('Superman')
    expect( mockedUseNavigate ).toHaveBeenCalledWith("?q=Superman");
    // screen.debug();
    // expect( input.value ).toBe('Batman')
  })
  
  
})
