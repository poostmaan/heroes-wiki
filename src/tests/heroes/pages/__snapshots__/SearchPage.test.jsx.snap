// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search Page /> testing should render correctly 1`] = `
<div>
  <div
    class="flex flex-row animate__animated animate__fadeIn"
  >
    <div
      class="basis-1/2 m-4"
    >
      <form
        method="GET"
        role="form"
      >
        <div
          class="relative "
        >
          <div
            class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
          >
            <svg
              aria-hidden="true"
              class="w-5 h-5 text-gray-500 dark:text-gray-400"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <input
            class="block p-4 pl-10 w-full text-sm text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
            id="default-search"
            name="searchText"
            placeholder="Search Mockups, Logos..."
            required=""
            type="text"
            value=""
          />
          <button
            class="text-white absolute right-2.5 bottom-2.5 bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-4 py-2 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
            type="submit"
          >
            Search
          </button>
        </div>
      </form>
    </div>
    <div
      class="basis-1/2 m-4"
    >
      <div
        class="p-4 mb-4 text-sm text-blue-700 bg-blue-100 rounded-lg dark:bg-blue-200 dark:text-blue-800"
        role="alert"
      >
        <span
          class="font-medium"
        >
          Hey!
        </span>
         Seach a superhero right here!
      </div>
      <div
        aria-label="alert-nofound"
        class="p-4 mb-4 text-sm text-red-700 bg-red-100 rounded-lg dark:bg-red-200 dark:text-red-800"
        role="alert"
        style="display: none;"
      >
        <span
          class="font-medium"
        >
          Oops!
        </span>
         There's no any superhero with that name
      </div>
    </div>
  </div>
</div>
`;
